lib/state/wallet.ts

```diff
@@ -0,0 +1,340 @@
+  /**
+   * The selected address of the current wallet.
+   */
+  selectedAddress?: string;
+  /**
+   * The active account of the current wallet.
+   */
+  activeAccount?: WalletAccount;
+  /**
+   * Select a wallet.
+   * @param wallet the selected wallet id or instance
+   * @returns void
+   */
+  selectWallet: (wallet: Wallet | string) => void;
+  /**
+   * Select an address.
+   * @param account the address to select
+   * @returns void
+   */
+  selectAddress: (account: string) => void;
+  /**
+   * Get the active signer for transactions.
+   * @returns KeyringPairOrExtSigner | null
+   */
+  getActiveSigner: () => KeyringPairOrExtSigner | null;
+  /**
+   * Disconnect the wallet.
+   * @returns void
+   */
+  disconnectWallet: () => void;
+};
+
+/**
+ * State type of user wallet config.
+ */
+export type WalletUserConfig = Partial<{
+  walletId: string;
+  selectedAddress: string;
+}>;
+
+/**
+ * State type of wallet.
+ */
+export type WalletState = {
+  /**
+   * Whether the wallet is connected.
+   */
+  connected: boolean;
+  /**
+   * Instance of the current wallet.
+   */
+  wallet?: Wallet;
+  /**
+   * The accounts of the current wallet.
+   */
+  accounts: WalletAccount[];
+  /**
+   * Error messages of the wallet.
+   */
+  errorMessages: WalletErrorMessage[];
+};
+
+/**
+ * Atom proxy storage.
+ * Used to access and write all atom state in the app.
+ */
+const store = getDefaultStore();
+
+/**
+ * Atom proxy storage of wallet state.
+ */
+const walletAtom = atom<WalletState>({
+  connected: false,
+  wallet: undefined,
+  accounts: [],
+  errorMessages: [],
+});
+
+/**
+ * Atom proxy storage of user wallet config.
+ * Stores the selected wallet and address in localStorage.
+ *
+ * @warning - when adding migrations all previous migrations in the list will have to left in place.
+ */
+const userConfigAtom = persistentAtom<WalletUserConfig>({
+  store,
+  key: "wallet-user-config",
+  defaultValue: {},
+  migrations: [
+    (state: unknown): WalletUserConfig => {
+      /**
+       * Migrate existing localStorage values to new atom state.
+       * So existing users don't have to reselect their wallet and address.
+       */
+      if (!state || Object.keys(state).length === 0) {
+        const walletId = globalThis.localStorage?.getItem("walletId");
+        const selectedAddress =
+          globalThis.localStorage?.getItem("accountAddress");
+
+        globalThis.localStorage?.removeItem("walletId");
+        globalThis.localStorage?.removeItem("accountAddress");
+
+        return {
+          walletId,
+          selectedAddress,
+        };
+      }
+
+      return state;
+    },
+  ],
+});
+
+/**
+ * Type of wallet errors.
+ */
+export type WalletErrorMessage = {
+  extensionName: string;
+  message: string;
+};
+
+/**
+ * List of supported wallets.
+ */
+export const supportedWallets = [
+  new PolkadotjsWallet(),
+  new SubWallet(),
+  new TalismanWallet(),
+];
+
+let accountsSubscriptionUnsub: VoidFunction | null = null;
```

Comment:
Does this need to be here? Or just inside `enableWallet `

Reply:
No this needs to be here since `enableWallet` needs to clear any existing subscription when another wallet is enabled.